### PR TITLE
Contributor system

### PR DESCRIPTION
**Pull request by mekolat**
This PR adds an account-wide #Contributor variable and restricts the use of contributor shirts.

I also want to add more incentive for players to contribute to the game so I want to add more contributor-only zones to the game. At the moment, there is only one contributor-only area (behind the bar in the Sanguine Vault). The contributor zones will be hangout spots for people that help by contributing to the game. No new map will be created, contributors will be able to access some previously inaccessible parts of existing maps (but providing no gameplay advantage).

There is 4 contributor levels:
* level 1: helper, tester or wiki editor
* level 2: dev (script, art, audio)
* level 3: GM or TMWC member
* level 4: admin

I have no plan at the moment for contributor levels but added them in case we need to differentiate the type of contributor in the future.

----
I already have ideas for 2 new contributor areas, will update this PR when it's ready

### CODE REVIEW

File: world/map/npc/functions/contributor.txt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    if(attachrid($@contributor_caster) != 0) goto L_Menu;
+    callsub S_Clean;
+    end;
+
+L_Menu:
+    if(getgmlevel() < 99) end;
+    mes "[Contributor Setup]";
+    mes "Target: "+ $@contributor_target$;
+    mes "Contributor level: "+ $@contributor_level;
+    menu
```
Comment (mekolat) [inline world/map/npc/functions/contributor.txt]: I have no idea why but the menu is only displayed half of the time when casting #contributor with a target

**Comment by CandyCurt** [comment]
what is wrong with current gm system? this feels unneccessarily redundant. if worries are regarding the gm tab - at least currently gm tab is non - functional for a gm level below 60

**Comment by mekolat** [comment]
@CandyCurt this is false; When I was given gm level 20 I could see any message sent with @wgm and I made further tests in local and saw that at gm 1+ you get gm messages

**Comment by Cassy1** [comment]
@mekolat huh? I can't see anything with my gm level 40.

**Comment by wushin** [comment]
binding the contributors shirt to a Account var. That does get around having to give randoms GM powers and whether or not they will abuse them and stops them from equipping, but still not selling or trading.

**Comment by mekolat** [comment]
@wushin since this PR restricts the use of the contributor shirt they will become worthless if the player that purchase them does not have the account var.

File: world/map/conf/magic-misc.sex [hunk 2]
```diff
@@ -1,6 +1,15 @@
 "--------------------------------------------------------------------------------"
 "Special-purpose quasispells"
 "--------------------------------------------------------------------------------"
+(SPELL (NONMAGIC) contributor "#contributor" (PC target)
+    (=>
+        (GUARD
```
Comment (wushin) [inline world/map/conf/magic-misc.sex]: I thought we could GUARD magic with GM level.

File: world/map/npc/functions/contributor.txt (hunk 1 above)
Comment (wushin) [inline world/map/npc/functions/contributor.txt]: add a debugmes on the attachrid($@contributor_caster) != 0 or check illia for use case.

File: world/map/conf/magic-misc.sex (hunk 2 above)
Comment (mekolat) [inline world/map/conf/magic-misc.sex]: I tried but I have no idea how.

**Pull request closed without merging**